pkg/openstack/loadbalancer.go

```diff
@@ -1324,37 +1554,50 @@ func (lbaas *LbaasV2) checkService(service *corev1.Service, nodes []*corev1.Node
 			}
 		}
 
-		if floatingSubnetID == "" {
-			floatingSubnetID = getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingSubnetID, lbaas.opts.FloatingSubnetID)
+		if !floatingSubnet.Configured() {
+			annos := floatingSubnetSpec{}
+			annos.subnetID = queryStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingSubnetID)
+			if annos.subnetID == "" {
+				annos.subnet = queryStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingSubnet)
+				annos.subnetTags = queryStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingSubnetTags)
+			}
+			if annos.Configured() {
+				floatingSubnet = annos
+			}
+		}
 
-			if floatingSubnetID == "" {
-				floatingSubnetName := getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingSubnet, "")
-				if floatingSubnetName != "" {
-					lbSubnet, err := lbaas.getSubnet(floatingSubnetName)
-					if err != nil || lbSubnet == nil {
-						klog.Warningf("Failed to find floating-subnet-id for Service %s: %v", serviceName, err)
-					} else {
-						floatingSubnetID = lbSubnet.ID
-					}
-				}
+		// apply defaults from CCM config
+		if floatingNetworkID == "" {
+			floatingNetworkID = lbaas.opts.FloatingNetworkID
+		}
+		if !floatingSubnet.Configured() {
```

Comment:
This can be combined with Line1557

Reply:
I'll do this, but it makes the control flow less clear, because it mashes up the various defaulting steps.